modules/tinymce/src/core/main/ts/api/html/DomParser.ts

```diff
@@ -554,7 +535,10 @@ const DomParser = (settings?: DomParserSettings, schema = Schema()): DomParser =
     };
 
     const rootNode = new AstNode(args.context || settings.root_name, 11);
-    builder(rootNode, html, settings, schema, args);
+    // The settings object we pass to purify is completely ignored, because we called setConfig earlier, but it makes the type signatures work
+    const element = purify.sanitize(html, { RETURN_DOM: true });
+    transferChildren(rootNode, element);
+    simplifyDom(rootNode, schema, settings, args);
     const state = walker(rootNode, settings, schema, nodeFilters, attributeFilters);
```

Comment:
I'm not fond that this is walking the entire dom 3 times... from the looks of things this could be done in 2 (or maybe less) walks instead so we'd be better doing that unless I'm missing a reason as to why we can't?